datafusion/physical-plan/src/aggregates/group_values/multi_group_by/mod.rs

```diff
@@ -251,12 +284,27 @@ struct VectorizedOperationBuffers {
 
 impl VectorizedOperationBuffers {
     fn clear(&mut self) {
-        self.append_row_indices.clear();
+        self.clear_append_row_indices();
         self.equal_to_row_indices.clear();
         self.equal_to_group_indices.clear();
         self.equal_to_results.clear();
         self.remaining_row_indices.clear();
     }
+
+    fn add_append_row_index(&mut self, row: usize) {
```

Comment:
It would probably faster and somewhat cleaner to do this in a single check on `append_row_indices` in `vectorized_append`,

something like:
```
append_row_indices.windows(2).all(|[x, y]| x + 1 == y)
```

will do.

Reply:
> I think even better, it should also be possible to just check the length of `append_row_indices`. If this is equal to the size of the incoming batch, all of the indices should be consecutive already (i.e. all values are unique), so it becomes a really cheap check.

but what if we only need to add the last half of the column. then the length would not be equal but it is still consecutive